src/v2/Apps/Order/Routes/Shipping/index.tsx

```diff
@@ -94,46 +94,41 @@ export interface ShippingProps extends SystemContextProps {
   isCommittingMutation: boolean
 }
 
-export interface ShippingState {
-  shippingOption: CommerceOrderFulfillmentTypeEnum
-  address: Address
-  phoneNumber: PhoneNumber
-  phoneNumberError: PhoneNumberError
-  phoneNumberTouched: PhoneNumberTouched
-  addressErrors: AddressErrors | {}
-  addressTouched: AddressTouched
-  selectedAddressID: string
-  saveAddress: boolean
-  shippingQuotes: ShippingQuotesType
-  shippingQuoteId?: string
-  savedAddressID?: string
-}
-
 const logger = createLogger("Order/Routes/Shipping/index.tsx")
-@track()
-export class ShippingRoute extends Component<ShippingProps, ShippingState> {
-  state: ShippingState = {
-    shippingOption: getShippingOption(
-      this.props.order.requestedFulfillment?.__typename
-    ),
-    // @ts-expect-error PLEASE_FIX_ME_STRICT_NULL_CHECK_MIGRATION
-    address: startingAddress(this.props.me, this.props.order),
-    addressErrors: {},
-    addressTouched: {},
-    savedAddressID: undefined,
-    phoneNumber: startingPhoneNumber(this.props.me, this.props.order),
-    phoneNumberError: "",
-    phoneNumberTouched: false,
-    selectedAddressID: defaultShippingAddressIndex(
-      this.props.me,
-      this.props.order
-    ),
-    shippingQuotes: getShippingQuotes(this.props.order),
-    shippingQuoteId: getSelectedShippingQuoteId(this.props.order),
-    saveAddress: true,
-  }
 
-  get touchedAddress() {
+export const ShippingRoute: React.FC<ShippingProps> = props => {
+  const [shippingOption, setShippingOption] = useState<
+    CommerceOrderFulfillmentTypeEnum
+  >(getShippingOption(props.order.requestedFulfillment?.__typename))
+  const [address, setAddress] = useState<Address>(
+    startingAddress(props.me, props.order) as Address
+  )
+  const [addressErrors, setAddressErrors] = useState<AddressErrors | {}>({})
+  const [addressTouched, setAddressTouched] = useState<AddressTouched>({})
+  const [savedAddressID, setSavedAddressID] = useState<string | undefined>(
+    undefined
+  )
+  const [phoneNumber, setPhoneNumber] = useState<PhoneNumber>(
+    startingPhoneNumber(props.me, props.order)
+  )
+  const [phoneNumberError, setPhoneNumberError] = useState<PhoneNumberError>("")
+  const [phoneNumberTouched, setPhoneNumberTouched] = useState<
+    PhoneNumberTouched
+  >(false)
+  const [selectedAddressID, setSelectedAddressID] = useState<string>(
+    defaultShippingAddressIndex(props.me, props.order)
+  )
+  const [shippingQuotes, setShippingQuotes] = useState<ShippingQuotesType>(
+    getShippingQuotes(props.order)
+  )
+  const [shippingQuoteId, setShippingQuoteId] = useState<string | undefined>(
+    getSelectedShippingQuoteId(props.order)
+  )
```

Comment:
If we're spending time moving this to `useState`, why not just use formik? It does all of this for us for free.

Reply:
Glad you mentioned this! The purpose of moving this to a FC is so we can consolidate all our address forms using formik. There is a [work-in-progress ticket](https://artsyproduct.atlassian.net/browse/NX-2993) that is already using formik on all of the others FC address forms, except for Shipping and PaymentPicker which need to be converted.
So, indeed, we have an immediate follow-up task to use formik and remove most of the code that is validating address "manually".

Reply:
Sweet 🙏